Flywoo H743 Pro (#27785)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/FlywooH743Pro/README.md

# Flywoo H743 Pro Flight Controller

The Flywoo H743 Pro is a flight controller produced by [Flywoo](https://www.flywoo.net/).

## Features

 - MCU - STM32H743 32-bit processor running at 480 MHz
 - IMU - Dual ICM42688
 - Barometer - SPL06
 - OSD - AT7456E
 - Onboard Flash: 500MByte
 - 7x UARTs
 - 13x PWM Outputs (12 Motor Output, 1 LED)
 - Battery input voltage: 2S-6S
 - BEC 3.3V 0.5A
 - BEC 5V 3A

>  - BEC 10V 3A

Reviewer: ```suggestion
 - BEC 10V 3A for video, gpio controlled
 - Dual switchable camera inputs
```

## Pinout

![Flywoo H743 Pro Board Top](Top.png "Flywoo H743 Pro Top")
![Flywoo H743 Pro Board Bottom](Bottom.png "Flywoo H743 Pro Bottom")

## UART Mapping

The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
receive pin for UARTn. The Tn pin is the transmit pin for UARTn.

 - SERIAL0 -> USB
 - SERIAL1 -> UART1 (RX/SBUS, DMA-enabled)
 - SERIAL2 -> UART2 (GPS, DMA-enabled)

>  - SERIAL3 -> UART3

Reviewer: ```suggestion
 - SERIAL3 -> UART3 (User)
```

 - SERIAL4 -> UART4 (GPS, DMA-enabled)
 - SERIAL6 -> UART6 (ESC Telemetry)

>  - SERIAL7 -> UART7

Reviewer: ```suggestion
 - SERIAL7 -> UART7 (User)
```

>  - SERIAL8 -> UART8 (DJI/MSP, DMA-enabled)

Reviewer: ```suggestion
 - SERIAL8 -> UART8 (DisplayPort, DMA-enabled)
```

> ## RC Input

Reviewer: replace RC section with this:
RC input is configured by default via the USART1 RX input. It supports all unidirectional RC protocols except PPM. 

Note: If the receiver is FPort the receiver must be tied to the USART1 TX pin ,  RSSI_TYPE set to 3, and SERIAL2_OPTIONS must be set to 7 (invert TX/RX, half duplex). For full duplex like CRSF/ELRS use both RX1 and TX1 and set RSSI_TYPE also to 3.

RC input is configured on the R1 (UART1_RX) pin. It supports all serial RC
protocols. For protocols requiring half-duplex serial to transmit
telemetry (such as FPort) you should setup SERIAL6 as an RC input serial port,
with half-duplex, pin-swap and inversion enabled.
 
## FrSky Telemetry
 

> FrSky Telemetry is supported using the T5 pin (UART5 transmit). You need to set the following parameters to enable support for FrSky S.PORT

Reviewer: ```suggestion
FrSky Telemetry is supported using  and unused UART,such as the T3 pin (UART3 transmit). You need to set the following parameters to enable support for FrSky S.PORT
```

>   - SERIAL5_PROTOCOL 10

Reviewer: ```suggestion
  - SERIA3_PROTOCOL 10
```

>   - SERIAL5_OPTIONS 7

Reviewer: ```suggestion
  - SERIAL3_OPTIONS 7
```

  
## OSD Support

> The Flywoo H743 Pro supports OSD using OSD_TYPE 1 (MAX7456 driver).

Reviewer: ```suggestion
The Flywoo H743 Pro supports OSD using OSD_TYPE 1 (MAX7456 driver) and simultaneously DisplayPort using TX8/RX8 on the HD VTX connector.
```

## PWM Output

> The Flywoo H743 Pro supports up to 13 PWM outputs. The pads for motor output

Reviewer: ```suggestion
The Flywoo H743 Pro supports up to 13 PWM or DShot outputs. The pads for motor output
```

M1 to M8 are provided on both the motor connectors and on separate pads, plus
M9-13 on a separate pads for LED strip and other PWM outputs.

The PWM is in 4 groups:

 - PWM 1-2   in group1
 - PWM 3-6   in group2
 - PWM 7-10  in group3
 - PWM 11-12 in group4
 - PWM 13    in group5

Channels within the same group need to use the same output rate. If
any channel in a group uses DShot then all channels in the group need
to use DShot. Channels 1-10 support bi-directional dshot.

## Battery Monitoring

> The board has a builting voltage and current sensor. The current

Reviewer: ```suggestion
The board has a built-in voltage sensor and external current sensor input. The current
```

sensor can read up to 130 Amps. The voltage sensor can handle up to 6S
LiPo batteries.

The correct battery setting parameters are:

 - BATT_MONITOR 4
 - BATT_VOLT_PIN 11
 - BATT_CURR_PIN 13
 - BATT_VOLT_MULT 11.1
 - BATT_AMP_PERVLT 40

## Compass

The Flywoo H743 Pro does not have a builtin compass, but you can attach an external compass using I2C on the SDA and SCL pads.

## Loading Firmware

Initial firmware load can be done with DFU by plugging in USB with the
bootloader button pressed. Then you should load the "with_bl.hex"
firmware, using your favourite DFU loading tool.

Once the initial firmware is loaded you can update the firmware using
any ArduPilot ground station software. Updates should be done with the

> *.apj firmware files.

Reviewer: ```suggestion
\*.apj firmware files.
```
